src/workerd/api/sockets.c++

```diff
@@ -9,8 +9,41 @@
 namespace workerd::api {
 
 
+bool isValidAddress(kj::StringPtr address) {
+  if (address.size() > (255 + 6)) {
+    // RFC1035 states that maximum domain name length is 255 octets. But we also add 6 to support
+    // port numbers in the address.
+    //
+    // IP addresses are always shorter, so we take the max domain length instead.
+    return false;
+  }
+
+  for (int i = 0; i < address.size(); i++) {
+    switch (address[i]) {
+      case '-':
+      case '.':
+      case ':':
+      case '_':
+      case '/': // To enable "unix:/" addresses (only used for testing, we expect oxy to reject this).
+        break;
+      default:
+        if (!std::isalnum(address[i])) {
+          // Character not in {'a'..'z', 'A'..'Z', '0'..'9'}
+          return false;
+        }
+        break;
+    }
+  }
+  return true;
+}
+
 jsg::Ref<Socket> connectImplNoOutputLock(
     jsg::Lock& js, jsg::Ref<Fetcher> fetcher, kj::String address) {
+
+  if (!isValidAddress(address)) {
+    JSG_FAIL_REQUIRE(TypeError, "Specified address contains unsupported characters.");
```

Comment:
This error message is a bit misleading if the address fails the length check but otherwise contains all valid characters.